src/modules/commander/commander.cpp

```diff
@@ -1056,7 +1056,10 @@ Commander::handle_command(vehicle_status_s *status_local,
 	break;
 	case vehicle_command_s::VEHICLE_CMD_CONTROL_HIGH_LATENCY: {
 			// only send the acknowledge from the commander, the command actually is handled by each mavlink instance
-			cmd_result = vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED;
+			// only send the acknowledge if the command is received from an external source
```

Comment:
The acknowledge is handled by the commander. The reason is that in that way transmitting multiple acknowledges is avoided.

Reply:
Sure, but you also lose the ability to even potentially acknowledge in a meaningful way. 

Even if you only sent the acknowledge in the mavlink task that's in iridium mode at least you'd get no response when Iridium is disabled instead of a meaningless acceptance.

Reply:
Which is also not ideal because then QGC will be resending the command multiple times. What I could do here is to check if any high_latency telemetry exists and decide based on that if the command is accepted or denied. Would that be acceptable?

The missing piece here would be that it is not checked if a correct mavlink instance exists. But if you check it on mavlink you can also not check if any iridium driver is started so it is also not completely safe.

Reply:
This should be moved to the MAVLink module and acknowledged there. In terms of retransmission: If Iridium is not active, it should send a NACK and that will tell QGC that it is not supported.